base/path.jl

```diff
@@ -125,6 +125,23 @@ abspath(a::AbstractString, b::AbstractString...) = abspath(joinpath(a,b...))
     end
 end
 
+@windows_only longpath(path::AbstractString) = longpath(utf16(path))
+@windows_only function longpath(path::UTF16String)
+    p = UInt32((sizeof(path)>>2) + 1)
+    while true
+        buflength = p
+        buf = zeros(UInt16,buflength)
+        p = ccall((:GetLongPathNameW, "Kernel32"), stdcall,
+            UInt32, (Cwstring, Ptr{UInt16}, UInt32),
+            path, buf, buflength)
+        systemerror(:longpath, p == 0)
+        if (p < buflength)
+            resize!(buf, p+1)
```

Comment:
If you are using `resize!` here, should probably initialize `buf = Array(UInt16, buflength)` outside the loop.   (It is confusing to use `zeros` if the initial contents of `buf` don't matter.)